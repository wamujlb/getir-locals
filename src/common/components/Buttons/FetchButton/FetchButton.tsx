import React from 'react';
import {
  Button,
  ButtonProps,
  ButtonTypeMap,
  CircularProgress,
} from '@mui/material';

export type FetchButtonProps = {
  isFetching?: boolean;
} & ButtonProps<ButtonTypeMap['defaultComponent'], {}>;

const FetchButton = React.memo(
  ({ isFetching, startIcon, ...rest }: FetchButtonProps) => {
    return (
      <Button
        startIcon={
          !isFetching ? (
            startIcon
          ) : (
            <CircularProgress size={18} color="inherit" />
          )
        }
        {...rest}
      />
    );
  }
);

FetchButton.displayName = 'FetchButton';

export default FetchButton;
