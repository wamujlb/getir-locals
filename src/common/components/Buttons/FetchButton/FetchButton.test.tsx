import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ShoppingCartIcon from 'common/components/icons/ShoppingCart/ShoppingCart';

import FetchButton from './FetchButton';

describe('#FetchButton', () => {
  afterAll(() => {
    cleanup();
  });

  it('should match the snapshot when isFetching is true', async () => {
    const screen = render(<FetchButton isFetching>button text</FetchButton>);

    expect(screen.container).toMatchSnapshot();
  });

  it('should match the snapshot when isFetching is false and startIcon is provided', async () => {
    const screen = render(
      <FetchButton startIcon={<ShoppingCartIcon />}>button text</FetchButton>
    );

    expect(screen.container).toMatchSnapshot();
  });

  it('should match the snapshot when isFetching is false and startIcon is not provided', async () => {
    const screen = render(<FetchButton>button text</FetchButton>);

    expect(screen.container).toMatchSnapshot();
  });
});
