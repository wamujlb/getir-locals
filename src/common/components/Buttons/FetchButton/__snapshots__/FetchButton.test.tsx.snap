// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#FetchButton should match the snapshot when isFetching is false and startIcon is not provided 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    button text
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`#FetchButton should match the snapshot when isFetching is false and startIcon is provided 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3.9,7.8h16.2V22H3.9V7.8z M9.1,1.9h5.8L16,3.1v4.7l-1.2,0V3.1H9.2v4.7L8,7.8V3.1L9.1,1.9z"
        />
      </svg>
    </span>
    button text
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`#FetchButton should match the snapshot when isFetching is true 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
        role="progressbar"
        style="width: 18px; height: 18px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </span>
    button text
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
